src/ccache/argprocessing.cpp

```diff
@@ -524,6 +524,18 @@ process_option_arg(const Context& ctx,
     }
   }
 
+  // emscriptcen's --em-config requires an argument.
+  if (arg == "--em-config") {
+    if (i == args.size() - 1) {
+      LOG("Missing argument to {}", args[i]);
+      return Statistic::bad_compiler_arguments;
+    }
+    state.common_args.push_back(args[i]);
+    ++i;
+    state.common_args.push_back(args[i]);
+    return Statistic::none;
+  }
```

Comment:
When a compiler option takes an argument and the argument doesn't need to be tracked or rewritten, you can add an entry like `{"--em-config", TAKES_ARG}` to the `compopts` table in `src/ccache/compopt.cpp` instead. Could you try if that works?

Reply:
Thanks. It works.